sample/trace_key.go

```diff
@@ -42,16 +43,29 @@ func (d *traceKey) build(trace *types.Trace) string {
 	// if they happen multiple times.
 	fieldCollector := make(map[string][]string)
 
+	// once a key gets this many unique values, it's off the charts in terms of uniqueness
+	// so we just stop looking for more.
+	// This is a safety valve to prevent us from someone sending a high-cardinality field
+	// in a giant trace.
+	const maxKeyLength = 100
 	// for each field, for each span, get the value of that field
 	spans := trace.GetSpans()
+	uniques := generics.NewSetWithCapacity[string](maxKeyLength)
+outer:
 	for _, field := range d.fields {
 		for _, span := range spans {
 			if val, ok := span.Data[field]; ok {
-				fieldCollector[field] = append(fieldCollector[field], fmt.Sprintf("%v", val))
+				v := fmt.Sprintf("%v", val)
+				uniques.Add(v)
+				if len(uniques) >= maxKeyLength {
+					break outer
```

Comment:
We should log that the key exceeded the configured max so we know it's stopped looking for additional keys.

Reply:
OK; the function here doesn't have access to a logger, so I had it return the key size it's using and log if it is the max value.

It occurs to me that we could now make a metric out of this, but I don't have time to do that. Please feel free if you think that's a good idea.

Reply:
Created a follow-up issue to track adding in the future if we decide to do it.